packages/components/resource-pages/bucket-policy-details-page.tsx

```diff
@@ -0,0 +1,231 @@
+            <dt>{t('Status')}</dt>
+            <dd>
+              <BPStatus obj={obj} />
+            </dd>
+            <dt>{t('Policy type')}</dt>
+            <dd>{policyType}</dd>
+            <dt>{t('Data sources')}</dt>
+            <dd>
+              {!!dataResourceCount ? (
+                <DataResourcesPopOver
+                  label={dataResourceLabel}
+                  dataResources={dataResources}
+                />
+              ) : (
+                dataResourceLabel
+              )}
+            </dd>
+            <dt>{t('ObjectBucketClaims')}</dt>
+            <dd>
+              {!!obcCount ? (
+                <OBCPopOver label={obcLabel} obcDetails={bucketPolicyOBCList} />
+              ) : (
+                obcLabel
+              )}
+            </dd>
+            <dt>{t('Replication buckets')}</dt>
+            <dd>
+              {!!repCount ? (
+                <OBCPopOver label={repLabel} obcDetails={replicationOBCList} />
+              ) : (
+                repLabel
+              )}
+            </dd>
+          </div>
+        </div>
+      </div>
+      {!!obcCount && (
+        <GetSecret
+          obj={{
+            metadata: {
+              name: bucketPolicyOBCList[0].name,
+              namespace: bucketPolicyOBCList[0].ns,
+            },
+          }}
+        />
+      )}
+    </>
+  );
+};
+
+const kind = referenceForModel(NooBaaBucketClassModel);
+
+const BucketPolicyDetailsPage: React.FC<PageProps> = ({ match }) => {
+  const { t } = useTranslation();
+  const { resourceName: name } = match.params;
+  const namespace = DATA_FEDERATION_NAMESPACE;
+  const [resource, loaded, error] = useK8sWatchResource<BucketClassKind>({
+    kind,
+    name,
+    namespace,
+    isList: false,
+  });
+
+  const [Modal, modalProps, launchModal] = useModalLauncher();
+
+  const actions = React.useCallback(() => {
+    return (
+      <Kebab
+        toggleType="Dropdown"
+        launchModal={launchModal}
+        extraProps={{
+          resource,
+          resourceModel: NooBaaBucketClassModel,
+        }}
+        customKebabItems={(t) => ({
+          Delete: {
+            value: t('Delete bucket'),
+          },
+        })}
+      />
+    );
+  }, [launchModal, resource]);
+
+  const breadcrumbs = [
+    {
+      name: t('Data Federation'),
+      path: '/mcgms/cluster',
+    },
+    {
+      name: t('Buckets'),
+      path: '/mcgms/cluster/resource/noobaa.io~v1alpha1~BucketClass',
+    },
+    {
+      name: t('Bucket details'),
+      path: '',
+    },
+  ];
+
+  return (
+    <>
+      <Modal {...modalProps} />
+      <DetailsPage
+        loaded={loaded}
+        loadError={error}
+        breadcrumbs={breadcrumbs}
+        actions={actions}
+        resourceModel={NooBaaBucketClassModel}
+        resource={resource}
+        pages={[
+          {
+            href: '',
+            name: 'Details',
+            component: BPDetails as any,
+          },
+        ]}
+      />
+    </>
+  );
+};
```

Comment:
Lot of these things can be converted to a generic component for reuse.

Reply:
we already have generic `DetailsPage` component, which we are reusing... all other things happening here are component specific...